src/test/java/teammates/test/cases/ui/pagedata/StudentCommentsPageDataTest.java

```diff
@@ -0,0 +1,231 @@
+package teammates.test.cases.ui.pagedata;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.CommentAttributes;
+import teammates.common.datatransfer.CourseAttributes;
+import teammates.common.datatransfer.CourseRoster;
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.FeedbackSessionResponseStatus;
+import teammates.common.datatransfer.FeedbackSessionResultsBundle;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.Url;
+import teammates.test.cases.BaseTestCase;
+import teammates.ui.controller.StudentCommentsPageData;
+import teammates.ui.template.Comment;
+import teammates.ui.template.CommentsForStudentsTable;
+import teammates.ui.template.CoursePagination;
+import teammates.ui.template.FeedbackSessionRow;
+import teammates.ui.template.QuestionTable;
+import teammates.ui.template.FeedbackResponseComment;
+import teammates.ui.template.ResponseRow;
+
+public class StudentCommentsPageDataTest extends BaseTestCase{
+    private static DataBundle dataBundle = getTypicalDataBundle();
+    private static StudentCommentsPageData data;
+    private static CourseAttributes course1;
+    private static StudentAttributes student1;
+    private static InstructorAttributes instructor1;
```

Comment:
I don't really recommend naming variables as `*1`, especially if there's only one of them. You can name them as `sample*`, or even simply `*`.